src/data/data2d/Spectrum2D/getSlice.ts

```diff
@@ -104,7 +104,7 @@ export function getSlice(
 function initiateData(from: number, to: number, size: number): NmrData1D {
   return {
     x: xSequentialFillFromTo({ from, to, size }),
-    re: new Float64Array(length),
+    re: new Float64Array(size),
```

Comment:
How was `length` not reported as non-existing variable?

Reply:
it is strange !!!!  the `"strict": true` is in the tsconfig` but it is not reported

Reply:
Found it: it's one of the many available globals on `window`:
![image](https://github.com/cheminfo/nmrium/assets/2352663/a9a797ce-eee5-45cb-aa52-d00a111aefe8)

I'll add it to our [list of restricted globals](https://github.com/cheminfo/eslint-config-cheminfo-react/blob/main/base.js#L24-L31)

Reply:
It is possible that length is defined as a global variable somewhere, but I've tried searching for it and couldn't find where it's defined

Reply:
Alright, I'll mark it as resolved and merge this PR. Updating ESLint rules can be done in another PR. What do you think?

Reply:
Yes, of course